### ISSUE TITLE
Injector-certs-secret name is hard coded as vault-injector-certs 

### ISSUE THREAD

**Issue by sameeraksc** [hashicorp/vault-helm#605]
**Describe the bug**
there is a secret called "vault-injector-certs" in the template. so that name is hardcoded. we tried to deploy two vault injectors in the same namespace as two helm releases (vault-dev, vault-prod)with an external vault servers.  because of this is it cant be done.

i think name should be something like this : name: {{ template "vault.fullname" . }}-injector-certs  but now its just  "vault-injector-certs"

I have used 0.6.0 for prod and 0.15.0 for dev in the same namespace. this issue occurred when I tried to upgrade prod also to 0.15.0 with two injector replicas.

**To Reproduce**
1. Install chart for injectors only  (two releases in one namespace)
2. Run vault command
3. See error (vault logs, etc.)

Other useful info to include: vault pod logs, `kubectl describe statefulset vault` and `kubectl get statefulset vault -o yaml` output

Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: Secret "vault-injector-certs" in namespace "default" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "vault"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"

**Expected behavior**

**Environment**
* Kubernetes version: 
  * Distribution or cloud vendor (OpenShift, EKS, GKE, AKS, etc.):
  * Other configuration options or runtime services (istio, etc.):
* vault-helm version:

Chart values:

```yaml
# Paste your user-supplied values here (`helm get values <release>`).
# Be sure to scrub any sensitive values!
```

**Additional context**

**Comment by mballoni** [comment]
I believe it describes two issues:
 1. the hardcoded name
 2. the missing labels (app.kubernetes.io/instance, app.kubernetes.io/managed-by and app.kubernetes.io/name).
 
 I suspect about the number two because I cannot upgrade my helm instalation if I increase the injectors replica number, otherwise it fails with exactly the same issue described.

**Comment by sidewinder12s** [comment]
Ya, looks like the missing labels cause issues with gitops type tooling like ArgoCD as it'll perpetually diff between the labels that get added by the helm chart and the missing labels when the secret gets patched.

**Comment by tvoran** [comment]
Yep, the missing labels issue was raised in https://github.com/hashicorp/vault-k8s/issues/378

And yes, to the original question, we don't support multiple injectors in the same namespace yet. The secret name is non-configurable right now because it's [not configurable on the vault-k8s injector side](https://github.com/hashicorp/vault-k8s/blob/5f9ad83ed9c75a9a04b8dea7c2fec779dc212bfc/helper/cert/source_gen.go#L34-L37) yet, so it'll need to be made configurable there first.
